clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/ChainResolver.java

```diff
@@ -131,4 +133,65 @@ public ChainResolver(final OperationsCodec<K, V> codec, Expiry<? super K, ? supe
       return new ResolvedChain.Impl<>(chain, key, result, 0, expirationTime);
     }
   }
+
+  @SuppressWarnings("unchecked")
+  public Chain compact(Chain chain, long now) {
+    Map<K, PutOperation<K, V>> compacted = new HashMap<>(1);
```

Comment:
Why one? If it's always one, why not just having a variable?

Reply:
It will always be a 1-entry map, so by the messed up `HashMap` contract it should be a 2 here. Fixed (and added a comment).